src/main.js

```diff
@@ -37,6 +37,7 @@ require.config({
         // implementations (e.g. cloud-based storage).
         "fileSystemImpl"    : "filesystem/impls/filer/FilerFileSystem",
 
+        "caman"             : "thirdparty/caman/caman.full.min",
```

Comment:
The main reason for not using `<script>` blocks was that that would only load the script once each browser session. Basically, the image viewer would be loaded once per browser session, and as such so would our script. This caused major issues when there were multiple images as the script wouldn't run again to update the image in the canvas, thus leading to blank canvases for all images once a new image was viewed. Since CamanJS won't work without a canvas, we needed to move the code to [this function](https://github.com/timmoy/brackets/blob/7c744eb5675c7243d749c62a26c4cf07d9c80b41/src/editor/ImageViewer.js#L151-L188) so that everytime the image loads we'll create a canvas for that image. There were also side effects of breaking the colour swatch extension when using the `<script>` blocks, but I'm not sure why that was.

Also from the same bug, using the `<script>` blocks to load the library ended up causing issues with the way that the editor was loaded. I remember that it was an issue with `DOMContentLoaded` and using a CamanJS function to replace the image with a canvas that showed the `DOMContentLoaded` event never fired as well as noting that asynchronous loading was being used after seeing the order of my debug statements and how the CamanJS function tried to replace a null object. I experimented with trying to force it to load the function synchronously via callbacks, but failed as brackets seems to be able to disable it somehow?

After doing some searching on the full explanation, I found someone with a [similar bug](https://bugzilla.mozilla.org/show_bug.cgi?id=667227#c8) on bugzilla that dealt with iframes and `<script>` blocks failing to load. It's not too related to the bug that happened here, but can be a valuable resource.